js/components/ChartForm.vue

```diff
@@ -73,73 +73,78 @@
+            <div class="help-block error-help">{{ errors.groupError }}</div>
           </div>
         </fieldset>
         <fieldset>
           <legend><span class="label label-default">{{ messages.targetLabel }}</span></legend>
           <div class="form-group targets" v-if="noGroup">
             <label class="control-label" for="target_count">{{ messages.targetCountLabel }}</label>
-            <input type="text" class="form-control" name="target_count" data-validate="targetDate" data-field='targets.No Groups' v-model.number="model.targets['No Groups']">
+            <input type="text" class="form-control" name="target_count" data-validate="validateTargetDate" data-field='targets.No Groups' v-model.number="model.targets['No Groups']">
           </div>
           <div class="form-group targets" v-else>
             <label>{{ messages.targetCountsLabel }}</label> <a data-toggle="collapse" role="button" :href="groupTargetsIdSelector" aria-expanded="true">{{ messages.actions.hideShow }}</a>
             <div class="collapse in" :id="groupTargetsId">
               <div class="well group-targets">
                 <div class="form-group form-group-sm" v-for="target in groupTargets">
                   <label class="control-label" :for="target.name">{{ target.label }}</label>
-                  <input type="text" class="form-control col-sm-10" :name="target.name" data-validate="targetDate" data-type='int' v-model.number="model.targets[target.label]">
+                  <input type="text" class="form-control col-sm-10" :name="target.name" data-validate="validateTargetDate" v-model.number="model.targets[target.label]">
                 </div>
                 <span class="target-total"><em>{{ messages.targetTotal }} {{ targetTotal }}</em></span>
               </div>
             </div>
           </div>
-          <div class="form-group">
+          <div class="form-group" :class="{ 'has-error': errors.endError }">
             <label class="control-label" for="target_date">{{ messages.targetEndDateLabel }}</label>
             <div class="input-group">
-              <input ref="targetEndDateInput" type="text" class="form-control vizr-date" name="target_date" data-field="end" data-v-model="targetEndDate" data-validate="date,targetDate" v-model="targetEndDate" @input="dateFieldChanged">
+              <input ref="targetEndDateInput" type="text" class="form-control vizr-date" name="target_date" data-field="end" data-v-model="targetEndDate" data-validate="validateDate,validateTargetDate" v-model="targetEndDate" @input="dateFieldChanged">
               <span class="input-group-addon" @click="showDatePicker($refs.targetEndDateInput)">
                 <span class="glyphicon glyphicon-calendar"></span>
               </span>
             </div>
-            <div class="help-block error-help">{{ targetEndDateErrors }}</div>
+            <div class="help-block error-help">{{ errors.endError }}</div>
           </div>
         </fieldset>
 
```

Comment:
Sometimes invalid fields aren't visible on the screen. This indicates why the save button is disabled.